flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java

```diff
@@ -143,11 +145,14 @@ public class OperationExecutor {
 
     private final ClusterClientServiceLoader clusterClientServiceLoader;
 
+    private final MaterializedTableManager materializedTableManager;
+
     @VisibleForTesting
     public OperationExecutor(SessionContext context, Configuration executionConfig) {
         this.sessionContext = context;
         this.executionConfig = executionConfig;
         this.clusterClientServiceLoader = new DefaultClusterClientServiceLoader();
+        this.materializedTableManager = new MaterializedTableManager(this);
```

Comment:
What do you see as the problem with circular references? I was mainly thinking about getting a global object because we needed to pass in conf information for the workflow scheduler for subsequent full refresh mode. Of course, this way also makes sense to me.

Reply:
I have changed MaterializedTableManager to as static util class.